src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -434,3 +480,202 @@ def get_pipeline_run_metadata(
+        """
+        # Extract region_name
+        region_match = re.search(r"sagemaker:(.*?):", pipeline_execution_arn)
+        region_name = region_match.group(1) if region_match else None
+
+        # Extract pipeline_name
+        pipeline_match = re.search(
+            r"pipeline/(.*?)/execution", pipeline_execution_arn
+        )
+        pipeline_name = pipeline_match.group(1) if pipeline_match else None
+
+        # Extract execution_id
+        execution_match = re.search(r"execution/(.*)", pipeline_execution_arn)
+        execution_id = execution_match.group(1) if execution_match else None
+
+        return region_name, pipeline_name, execution_id
+
+    def fetch_status(self, run: "PipelineRunResponse") -> ExecutionStatus:
+        """Refreshes the status of a specific pipeline run.
+
+        Args:
+            run: The run that was executed by this orchestrator.
+
+        Returns:
+            the actual status of the pipeline job.
+
+        Raises:
+            AssertionError: If the run was not executed by to this orchestrator.
+            ValueError: If it fetches an unknown state or if we can not fetch
+                the orchestrator run ID.
+        """
+        # Make sure that the stack exists and is accessible
+        if run.stack is None:
+            raise ValueError(
+                "The stack that the run was executed on is not available "
+                "anymore."
+            )
+
+        # Make sure that the run belongs to this orchestrator
+        assert (
+            self.id
+            == run.stack.components[StackComponentType.ORCHESTRATOR][0].id
+        )
+
+        # Initialize the Sagemaker client
+        session = self._get_sagemaker_session()
+        sagemaker_client = session.sagemaker_client
+
+        # Fetch the status of the _PipelineExecution
+        if METADATA_ORCHESTRATOR_RUN_ID in run.run_metadata:
+            run_id = run.run_metadata[METADATA_ORCHESTRATOR_RUN_ID].value
+        elif run.orchestrator_run_id is not None:
+            run_id = run.orchestrator_run_id
+        else:
+            raise ValueError(
+                "Can not find the orchestrator run ID, thus can not fetch "
+                "the status."
+            )
+        status = sagemaker_client.describe_pipeline_execution(
+            PipelineExecutionArn=run_id
+        )["PipelineExecutionStatus"]
+
+        # Map the potential outputs to ZenML ExecutionStatus. Potential values:
+        # https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/PipelineState
+        if status in ["Executing", "Stopping"]:
+            return ExecutionStatus.RUNNING
+        elif status in ["Stopped", "Failed"]:
+            return ExecutionStatus.FAILED
+        elif status in ["Succeeded"]:
+            return ExecutionStatus.COMPLETED
+        else:
+            raise ValueError("Unknown status for the pipeline execution.")
+
+    def generate_metadata(
```

Comment:
Feel free to ignore, but to me these methods don't generate something, they instead compute something.

Reply:
Nice idea, changed them to `compute_XXXX`.